polygeist/tools/cgeist/Lib/CGExpr.cc

```diff
@@ -1082,15 +1082,8 @@ llvm::Optional<sycl::SYCLMethodOpInterface> MLIRScanner::createSYCLMethodOp(
 
   // SYCLCastOps are abstracted to avoid missing method calls due to
   // implementation details.
-  if (const llvm::Optional<sycl::SYCLCastOp> Cast =
-          trackSYCLCast(OperandsCpy[0])) {
-    auto NewArg = (*Cast)->getOperand(0);
-    // Make sure the memory space is not changed:
-    const auto MemSpace =
-        OperandsCpy[0].getType().cast<MemRefType>().getMemorySpaceAsInt();
-    if (NewArg.getType().cast<MemRefType>().getMemorySpaceAsInt() != MemSpace) {
-      NewArg = castToMemSpace(NewArg, MemSpace);
-    }
+  if (Operation *Cast = trackCasts(OperandsCpy[0])) {
+    auto NewArg = Cast->getOperand(0);
```

Comment:
auto -> Operation *

Reply:
It is not a type, but a value. I'd say type is not needed here as `getOperand` is pretty popular in our codebase and the return type is quite straightforward.

Reply:
I did a grep on the MLIR codebase (for "=.*getOperand\(\);") and most instances declare the LHS with `Value` rather than `auto` (found 28 instances, 23 use `Value`). Just a data point. I realize we inherited code from cgeist where `auto` is used (too) pervasively, had `cgeist` been forced to upstream the code to the MLIR codebase I am confident reviewers would ask for changes.